scipy/optimize/tests/test_milp.py

```diff
@@ -271,3 +271,38 @@ def test_infeasible_prob_16609():
     res = milp(c, integrality=integrality, bounds=bounds,
                constraints=constraints)
     np.testing.assert_equal(res.status, 2)
+
+
+@pytest.mark.slow
+def test_milp_timeout_16545():
+    # Ensure solution is not thrown away if MILP solver times out
+    # -- see gh-16545
+    np.random.seed(16545)
+    A = np.zeros((100, 100))
+    for c in range(100):
+        for v in range(100):
+            A[c, v] = np.random.randint(0, 5)
+    b_lb = np.array([-1 * np.inf for c in range(100)])
+    b_ub = np.array([25 for c in range(100)])
+    constraints = LinearConstraint(A, b_lb, b_ub)
+    variable_lb = [0 for v in range(100)]
+    variable_ub = [1 for v in range(100)]
+    variable_bounds = Bounds(variable_lb, variable_ub)
+    integrality = [1 for v in range(100)]
+    c_vector = 100 * [-1]
```

Comment:
```suggestion
    rng = np.random.default_rng(5123833489170494244)
    A = rng.integers(0, 5, size=(100, 100))
    b_lb = np.full(100, fill_value=-np.inf)
    b_ub = np.full(100, fill_value=25)
    constraints = LinearConstraint(A, b_lb, b_ub)
    variable_lb = np.zeros(100)
    variable_ub = np.ones(100)
    variable_bounds = Bounds(variable_lb, variable_ub)
    integrality = np.ones(100)
    c_vector = -np.ones(100)
```
using `np.random.default_rng` instead of `randint` to make Pamphile happy.
Suggesting the rest for concision.